File: backend/src/main/java/com/example/backend/domain/admin/exception/AdminException.java
package com.example.backend.domain.admin.exception;

import org.springframework.http.HttpStatus;

public class AdminException extends RuntimeException {
        private final AdminErrorCode adminErrorCode;
        
    public AdminException(AdminErrorCode adminErrorCode) {
        super(adminErrorCode.getMessage());
        this.adminErrorCode = adminErrorCode;
    }

    public HttpStatus getStatus() {
        return adminErrorCode.getHttpStatus();
    }

    public String getCode() {
        return adminErrorCode.getCode();
    }
}

File: backend/src/main/java/com/example/backend/domain/admin/exception/AdminException.kt
package com.example.backend.domain.admin.exception;

import org.springframework.http.HttpStatus;

class AdminException(
    private val adminErrorCode: AdminErrorCode
) : RuntimeException(adminErrorCode.message) {

    override val message: String
        get() = adminErrorCode.message

    val status: HttpStatus
        get() = adminErrorCode.httpStatus

    val code: String
        get() = adminErrorCode.code
}
